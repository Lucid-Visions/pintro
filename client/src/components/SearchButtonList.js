import React from "react"
import {View, StyleSheet} from "react-native"
import SearchButton from "./SearchButton"
import searchIconsData from "../assets/searchIconsData"

class SearchButtonList extends React.Component {
    constructor() {
        super()
        this.state = {
            icons: searchIconsData,
        }
    }


    
    render() {
        const searchButtons = this.state.icons.map(item => <SearchButton callbackFunction={this.props.searchCallback} key={item.id} item={item} />)
        
        return (
            <View style={styles.container}>
                {searchButtons}
            </View>
        )    
    }

    
}

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        paddingHorizontal: 20
    }
});

export default SearchButtonList