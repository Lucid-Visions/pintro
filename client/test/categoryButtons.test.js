import React from 'react'
import { create, act } from 'react-test-renderer'
import SearchButtonList from '../src/components/SearchButtonList'

describe('<SearchButtonList />',() => {
    let selectedItems = []

    const mockCallback = jest.fn((category, isSelected) => {
        if(isSelected) selectedItems.push(category)
        else {
            const position = selectedItems.indexOf(category);

            if ( ~position ) selectedItems.splice(position, 1);
        }
    })

    beforeEach(() => {
        selectedItems = []
    })

    test('snapshot of the component', () => {
        const component = create(<SearchButtonList searchCallback={mockCallback} />).toJSON()
        expect(component).toMatchSnapshot()
    })

    test('test if selected category is added to array of selected categories', async () => {
        const component = create(<SearchButtonList searchCallback={mockCallback} />)
        const instance = component.root
        const buttons = instance.findByType('View')
        const categoryOne = buttons.children[0].children[0].props

        expect(selectedItems).toHaveLength(0)
        await categoryOne.onPress()
        expect(selectedItems).toHaveLength(1)
    })

    test('test adding and removing a few categories', async () => {
        const component = create(<SearchButtonList searchCallback={mockCallback} />)
        const instance = component.root
        const buttons = instance.findByType('View')
        const categoryOne = buttons.children[0].children[0].props
        const categoryTwo = buttons.children[1].children[0].props
        const categoryThree = buttons.children[2].children[0].props

        expect(selectedItems).toHaveLength(0)
        await categoryOne.onPress()
        expect(selectedItems).toHaveLength(1)
        await categoryOne.onPress()
        expect(selectedItems).toHaveLength(0)

        await categoryTwo.onPress()
        expect(selectedItems).toHaveLength(1)

        await categoryThree.onPress()
        expect(selectedItems).toHaveLength(2)
        await categoryThree.onPress()
        expect(selectedItems).toHaveLength(1)
    })
})