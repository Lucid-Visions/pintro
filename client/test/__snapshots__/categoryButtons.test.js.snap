// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchButtonList /> snapshot of the component 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 6,
        "marginRight": 6,
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 35,
          "height": 65,
          "overflow": "hidden",
          "width": 65,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "poppins-medium",
              "fontSize": 10,
              "marginVertical": 20,
            },
            false,
          ]
        }
      >
        PEOPLE
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 6,
        "marginRight": 6,
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 35,
          "height": 65,
          "overflow": "hidden",
          "width": 65,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "poppins-medium",
              "fontSize": 10,
              "marginVertical": 20,
            },
            false,
          ]
        }
      >
        COMPANIES
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 6,
        "marginRight": 6,
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 35,
          "height": 65,
          "overflow": "hidden",
          "width": 65,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "poppins-medium",
              "fontSize": 10,
              "marginVertical": 20,
            },
            false,
          ]
        }
      >
        CONTENT
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 6,
        "marginRight": 6,
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 35,
          "height": 65,
          "overflow": "hidden",
          "width": 65,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "poppins-medium",
              "fontSize": 10,
              "marginVertical": 20,
            },
            false,
          ]
        }
      >
        GROUPS
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 6,
        "marginRight": 6,
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 35,
          "height": 65,
          "overflow": "hidden",
          "width": 65,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "poppins-medium",
              "fontSize": 10,
              "marginVertical": 20,
            },
            false,
          ]
        }
      >
        SPACES
      </Text>
    </View>
  </View>
</View>
`;
